src/storage-types/src/connections.rs

```diff
@@ -1234,6 +1337,31 @@ impl RustType<ProtoTunnel> for Tunnel<InlinedConnection> {
     }
 }
 
+impl<C: ConnectionAccess> AlterCompatible for Tunnel<C> {
+    fn alter_compatible(
+        &self,
+        id: GlobalId,
+        other: &Self,
+    ) -> Result<(), crate::controller::StorageError> {
+        let compatible = match (self, other) {
+            (Self::Ssh(s), Self::Ssh(o)) => s.alter_compatible(id, o).is_ok(),
+            _ => true,
```

Comment:
Should this default to being allowed? Why do we single-out the ssh tunnels, when the implementation on `SshConnection` defaults to allowing everything anyways? Could you leave some comments explaining the reasoning?

Reply:
This isn't `SshConnection` it's for `SshTunnel` which might have a more complex compatibility check.

Reply:
I'll clarify with a comment once I sort this out.

Reply:
Converted the other matches to a strict equality comparison which is the default case for elements that themselves do not implement `alter_compatible`.